platform/iphone/view_controller.mm

```diff
@@ -29,60 +29,60 @@
 /*************************************************************************/
 
 #import "view_controller.h"
+#import "core/os/keyboard.h"
+#import "gl_view.h"
+#import "os_iphone.h"
+#import "video_iphone.h"
 
-#include "os_iphone.h"
+static String keyboard_text;
+static GodotGameViewController *_instance = nil;
 
-extern "C" {
+extern CGFloat _points_to_pixels(CGFloat points);
 
-int add_path(int, char **);
-int add_cmdline(int, char **);
+@interface GodotGameViewController ()
 
-int add_path(int p_argc, char **p_args) {
+@property(nonatomic, strong) NSMutableArray *activeTouches;
 
-	NSString *str = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"godot_path"];
-	if (!str)
-		return p_argc;
-
-	p_args[p_argc++] = "--path";
-	[str retain]; // memory leak lol (maybe make it static here and delete it in ViewController destructor? @todo
-	p_args[p_argc++] = (char *)[str cString];
-	p_args[p_argc] = NULL;
-
-	return p_argc;
-};
-
-int add_cmdline(int p_argc, char **p_args) {
-
-	NSArray *arr = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"godot_cmdline"];
-	if (!arr)
-		return p_argc;
-
-	for (int i = 0; i < [arr count]; i++) {
+@end
 
-		NSString *str = [arr objectAtIndex:i];
-		if (!str)
-			continue;
-		[str retain]; // @todo delete these at some point
-		p_args[p_argc++] = (char *)[str cString];
-	};
+@implementation GodotGameViewController
 
-	p_args[p_argc] = NULL;
+- (id)initWithFrame:(CGRect)frame {
+	if (self = [super init]) {
+		self.activeTouches = [NSMutableArray arrayWithCapacity:10];
+		self.view.frame = frame;
 
-	return p_argc;
-};
-}; // extern "C"
+		[[NSNotificationCenter defaultCenter]
+				addObserver:self
+				   selector:@selector(keyboardOnScreen:)
+					   name:UIKeyboardDidShowNotification
+					 object:nil];
 
-@interface ViewController ()
+		[[NSNotificationCenter defaultCenter]
+				addObserver:self
+				   selector:@selector(keyboardHidden:)
+					   name:UIKeyboardDidHideNotification
+					 object:nil];
 
-@end
+		_instance = self;
+	}
+	return self;
+}
 
-@implementation ViewController
+- (void)loadView {
```

Comment:
This will be called the first time any object attempts to access the `view` property of this `UIViewController`. Here we can programmatically create our view hierarchy.